web-registry/src/lib/ecocredit/api.ts

```diff
@@ -307,13 +313,31 @@ export const getBatchesByProjectWithSupply = async (
 const getClassProjectForBatch = async (
   batch: BatchInfo,
   sanityCreditClassData?: AllCreditClassQuery,
+  reactQueryClient?: QueryClient,
 ): Promise<ClassProjectInfo> => {
+  let metadata, projectData;
   const { projectId } = batch;
-  const { project } = await getProject(projectId);
-  let metadata;
+  if (reactQueryClient) {
+    projectData = await getFromCacheOrFetch({
+      query: getProjectQuery({
+        request: { projectId },
+      }),
+      reactQueryClient,
+    });
+  } else {
+    projectData = await getProject(projectId);
```

Comment:
Keeping the old path for retro-compatibility of the function. Once all paths use the react-query one we will be able to remove it.

Reply:
could you add a TODO?